src/Foundatio/Lock/CacheLockProvider.cs

```diff
@@ -22,24 +23,37 @@ public class CacheLockProvider : ILockProvider, IHaveLogger, IHaveTimeProvider
     private readonly AsyncLock _lock = new();
     private bool _isSubscribed;
     private readonly ILogger _logger;
+    private readonly ILoggerFactory _loggerFactory;
     private readonly Histogram<double> _lockWaitTimeHistogram;
     private readonly Counter<int> _lockTimeoutCounter;
+    private readonly IResiliencePolicyProvider _resiliencePolicyProvider;
+    private readonly IResiliencePolicy _releasePolicy;
+    private readonly IResiliencePolicy _isLockedPolicy;
+    private readonly IResiliencePolicy _renewPolicy;
 
-    public CacheLockProvider(ICacheClient cacheClient, IMessageBus messageBus, ILoggerFactory loggerFactory = null) : this(cacheClient, messageBus, null, loggerFactory) { }
+    public CacheLockProvider(ICacheClient cacheClient, IMessageBus messageBus, ILoggerFactory loggerFactory = null) : this(cacheClient, messageBus, null, null, loggerFactory) { }
 
-    public CacheLockProvider(ICacheClient cacheClient, IMessageBus messageBus, TimeProvider timeProvider, ILoggerFactory loggerFactory = null)
+    public CacheLockProvider(ICacheClient cacheClient, IMessageBus messageBus, TimeProvider timeProvider, IResiliencePolicyProvider resiliencePolicyProvider, ILoggerFactory loggerFactory = null)
     {
-        _timeProvider = timeProvider ?? cacheClient.GetTimeProvider();
-        _logger = loggerFactory?.CreateLogger<CacheLockProvider>() ?? cacheClient.GetLogger() ?? NullLogger<CacheLockProvider>.Instance;
+        _timeProvider = timeProvider ?? cacheClient.GetTimeProvider() ?? TimeProvider.System;
+        _loggerFactory = loggerFactory ?? cacheClient.GetLoggerFactory() ?? NullLoggerFactory.Instance;
+        _logger = _loggerFactory.CreateLogger<CacheLockProvider>();
         _cacheClient = new ScopedCacheClient(cacheClient, "lock");
         _messageBus = messageBus;
 
+        _resiliencePolicyProvider = resiliencePolicyProvider ?? cacheClient.GetResiliencePolicyProvider();
+        _releasePolicy = _resiliencePolicyProvider?.GetPolicy(nameof(ILockProvider.ReleaseAsync)) ?? new ResiliencePolicy(_timeProvider, _logger) { MaxAttempts = 15 };
```

Comment:
this attempts seems awefully high, also shouldn't there be some kind of backoff multiplier?

Reply:
It's the exact same as it was before with the attempts and the backoff.

Reply:
This was pretty crazy. No idea why we had it set like that, but I am changing it to use standard retry which is 3 attempts with exponential backoff.

Reply:
Sounds great